### ISSUE TITLE
Allow for dict keys to only be set if value is not null

### ISSUE THREAD

**Issue by alexgmin** [bhch/django-jsonform#18]
We have a field that is like this:
```
    action_timestamps = JSONField(
        default=dict,
        schema={
            'type': 'object',
            'keys': {
                'timestamp_1': {
                    'type': 'string',
                },
                'timestamp_2': {
                    'type': 'string',
                },
            }
        },
    )
```

Outside of the admin, we set the timestamps only when they are needed.
So for example this could be a valid value for us:
```
{
    "timestamp_1": "2018-11-19 12:10:02.353829"
}
```

However, once you edit the model from the admin, that value is set to:

```
{
    "timestamp_1": "2018-11-19 12:10:02.353829", 
    "timestamp_2": ""
}
```

Would it be possible to add a keyword to the schema key that makes so the key isn't stored unless it has a non falsy value?

**Comment by bhch** [comment]
Why can't you ignore the key if it has an empty value?

I think I can add a keyword called `skip_if_empty` which will omit the key from the final data, but I'd like to know some use-cases and reasons for this feature.

**Comment by alexgmin** [comment]
It's mostly about not having to store data that isn't needed, and that the admin is not the only source of inputs, so now we would have to replicate the behavior of the library also in other places if we want to keep everything standardized. 
The workaround isn't hard (just deleting keys with empty values in an extended model field), so you can close the issue if you don't think it's a good feature.

**Comment by bhch** [comment]
Fixed in [v2.10.0](https://github.com/bhch/django-jsonform/releases/tag/v2.10.0). A new argument called `pre_save_hook` can be used for processing json data before saving.
